File: src/70. Climbing Stairs.js
/**
 * @param {number} n
 * @return {number}
 */

function climbStairs(n) {
  let ones = 1
  let twos = 0
  for (let i = 1; i <= n - 1; i++) [ones, twos] = [ones + twos, ones]
  return ones + twos
}

File: src/70. Climbing Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        int ones = 1;
        int twos = 0;
        for (int i = 2; i <= n; ++i) {
            int sum = ones + twos;
            twos = ones;
            ones = sum;
        }
        return ones + twos;
    }
};